### PR TITLE
fix: always include MinimumFieldsToInclude

### PR DESCRIPTION
**Pull request by y-lakhdar**
https://coveord.atlassian.net/browse/JSUI-3525

`FieldsToInclude` property will always include minimal amount of fields, regardless of the options `data-fields-to-include` and `data-auto-select-fields-to-include`.

### CODE REVIEW

**Review by alexprudhomme — APPROVED** [review]
lgtm learned about https://underscorejs.org/

File: unitTests/ui/ResultListTest.ts [hunk 1]
```diff
@@ -569,6 +569,17 @@ export function ResultListTest() {
         );
       });
 
+      it('should also get the minimal amount of fields to include when the option is false', () => {
```
Comment (olamothe) [inline unitTests/ui/ResultListTest.ts:572]: Could add one test to verify that duplicate are removed

**Review by olamothe — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]
